Model/ExtensionNotification.php

```diff
@@ -128,12 +138,19 @@ private function checkExtensionVersion()
     private function getLatestVersionFromRepository()
     {
         if ($this->repoData === null) {
-            $json = file_get_contents(
-                self::REPOSITORY_URL,
-                false,
-                stream_context_create(self::CURL_OPT)
+            $response = $this->guzzleClient->request(
+                'GET',
+                self::REPOSITORY_URI,
+                [
+                    'headers' => self::APPLICATION_JSON_HEADER,
+                ]
             );
-            $this->repoData = json_decode($json);
+
+            if ($response->getStatusCode() != 200) {
+                throw new \Exception($response->getReasonPhrase());
+            }
+
+            $this->repoData = json_decode($response->getBody()->getContents());
```

Comment:
Magento has its own interface for (de)serialization (`Magento\Framework\Serialize\SerializerInterface`) with a specific implementation for JSON (de)serialization (`Magento\Framework\Serialize\Serializer\Json`)

Reply:
Love that this suggestion is coming from the Serializator.

Reply:
Thanks a lot for your suggestion @Serializator , I made the requested change ;)